packages/api/app/Controllers/Http/IdeaController.js

```diff
@@ -46,20 +49,15 @@ class IdeaController {
 			const { init, commit } = getTransaction();
 			trx = await init();
 
-			idea = await Idea.create(
-				{
-					title,
-					description,
-				},
-				trx,
-			);
+			idea = await Idea.create({ title, description }, trx);
 			await idea.user().associate(ideaOwner, trx);
 			if (keywords) {
 				await this.syncronizeTerms(trx, keywords, idea);
 			}
 			await idea.load('terms');
 
-			await Promise.all([Algolia.saveIndex('idea', idea), commit()]);
+			await commit();
+			await Algolia.handleObject.save(algoliaIndexName, await algoliaPopulatedQuery(idea.id));
```

Comment:
This is a good point for refactoring. We can move this to a background job in the future, but for now, it will be a simpler way to do this integration without much effort